crates/solvers/src/boundary/naive.rs

```diff
@@ -108,6 +113,15 @@ pub fn solve(
     let boundary_solution =
         multi_order_solver::solve(&slippage.context(), boundary_orders, &boundary_pool)?;
 
+    let nmb_orders = boundary_solution.trades().count();
```

Comment:
nit: This feels like it could be in the domain logic. I would avoid adding addition logic to `boundary` (it should shrink over time, and not grow). In fact, I would just change the `boundary` module to return a solution with a gas estimate and then move the score calculation to the `domain` (as this is IMO domain logic core to the solver engines). I would try and keep the `boundary` as "wrap-unwrap" as possible.

Reply:
Moved out.